wkcuber/wkcuber/utils.py

```diff
@@ -214,55 +218,57 @@ def add_batch_size_flag(parser: argparse.ArgumentParser) -> None:
 
 
 class BufferedSliceWriter(object):
+    """
+    The BufferedSliceWriter buffers multiple slices before they are written to disk.
+    The amount of slices that get buffered is specified by `buffer_size`.
+    As soon as the buffer is full, the data gets written to disk.
+
+    The user can specify along which dimension the data is sliced by using the parameter `dimension`.
+    To slice along the x-axis use `0`, for the y-axis use `1`, or for the z-axis use `2` (default: dimension=2).
+
+    This class is supposed to be used with a context manager (see example below).
+    If the buffer is non empty after the user finished writing (i.e. the number of written slices is not a multiple of `buffer_size`),
+    exiting the context will automatically write the buffer to disk.
+    Entering the context returns a generator with consumes slices (np.ndarray).
+    Note: this generator pattern requires to start the generator by sending `None` to it.
+
+    Usage:
+    data_cube = ...
+    with BufferedSliceWriter(...) as writer:
+        writer.send(None)  # to start the generator
+        for data_slice in data_cube:
+            writer.send(data_slice)
+
+    """
     def __init__(
         self,
-        dataset_path: Path,
-        layer_name: str,
-        dtype: np.dtype,
-        origin: Union[Tuple[int, int, int], List[int]],
+        view: View,
```

Comment:
I don't understand why I get an error when I specify the type to be `View` (instead of leaving it blank) (same for BufferedSliceReader). I assume that there is a circular dependency, but I explicitly only import `View` for typechecking.

Reply:
In this case you need to quote `"View"`, to ensure that the class is not needed when evaluating the code. During type inference this forward declaration is resolved. See https://stackoverflow.com/a/63170635/3581794
```suggestion
        view: "View",
```